Add errored and completed meta job run information

cds95 opened the pull request:
Adds errored and completed fields in the meta object of the response returned from `http://localhost:6688/v2/runs?jobSpecId=${JOB_ID}`.  A sample response is shown below.

```
{
    "data": [
       {
            ...jobSpecData
       }
    ],
    "meta": {
        "completed": "1",
        "count": 1,
        "errored": "0"
    }
}
```

cds95 commented on core/web/helpers.go:
```diff
@@ -33,6 +33,29 @@ func jsonAPIError(c *gin.Context, statusCode int, err error) {
+	err error,
+	meta map[string]string,
+) {
+	if errors.Cause(err) == orm.ErrorNotFound {
```
This function has a bit of duplicated logic with the function below it `paginatedResponse`.  The only difference is that this function calls `NewPaginatedResponseWithMeta` whilst the `paginatedResponse` calls `NewPaginatedResponse`.  I'm open to suggestions on how this can be improved.

samsondav commented:
@cds95 whats the story here, do you need a review?

cds95 commented:
@samsondav hey yes I do.  Should I leave a message somewhere if I have a PR ready for review in the future?  Sorry for the late reply.  I didn't see your comment.

samsondav reviewed (changes requested):
This looks fine. Can we see a test on the controller?

cds95 commented:
Sure.  I've added them and pushed to the PR.  I also added some extra changes as I uncovered an issue when running the tests.  The issue was that the completed and errored counts only counted what was fetched from the DB and did not count the total number of entries in the DB.  This led to incorrect counts when the `size` parameter was specified.  I've changed it so that the counts are obtained directly from the DB.

samsondav commented:
Approved, opened permissioned branch here: https://github.com/smartcontractkit/chainlink/pull/4187

samsondav commented:
#4187 is merged

samsondav commented:
#4187 is merged
